### PR TITLE
Implement serialization for UDWF and UDAF in plan protobuf

### LINKED ISSUE #4584: Implement serialization for UDAF as window function in plan protobuf
Follow-up of PR #4553 , serialization of UDAF as window function in plan protobuf is left as a todo, should be tracked in this issue.

### PR DESCRIPTION
**Pull request by parkma99**
# Which issue does this PR close?

Closes #4584 #6733 

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by parkma99** [comment]
cc @alamb I am not sure this is good implement?😊

File: datafusion/proto/src/logical_plan/mod.rs [hunk 1]
```diff
@@ -2786,12 +2786,66 @@ mod roundtrip_tests {
...
+        );
+
+        let test_expr5 = Expr::WindowFunction(expr::WindowFunction::new(
+            WindowFunction::AggregateUDF(Arc::new(dummy_agg.clone())),
+            vec![col("col1")],
+            vec![col("col1")],
+            vec![col("col2")],
             row_number_frame,
         ));
+        ctx.register_udaf(dummy_agg);
```
Comment (alamb) [inline datafusion/proto/src/logical_plan/mod.rs:2843]: THis is great!

Now all we need is a test for the `WindowUDF` (aka `WindowFunction::WindowUDF`) and I think this PR is ready to go!

File: datafusion/proto/src/logical_plan/to_proto.rs [hunk 2]
```diff
@@ -585,16 +585,16 @@ impl TryFrom<&Expr> for protobuf::LogicalExprNode {
                         )
                     }
                     // TODO: Tracked in https://github.com/apache/arrow-datafusion/issues/4584
```
Comment (alamb) [inline datafusion/proto/src/logical_plan/to_proto.rs:587]: Perhaps we can remove the TODO comments as well

**Review by alamb — COMMENTED** [review]
This looks really good to me -- thank you @parkma99 🤗

**Comment by parkma99** [comment]
Thanks @alamb , I will update the test case lately.😄

File: datafusion/proto/src/logical_plan/mod.rs [hunk 3]
```diff
@@ -2786,12 +2788,155 @@ mod roundtrip_tests {
...
+            ///
+            /// Note this is the SLOWEST, but simplest, way to evaluate a
+            /// window function. It is much faster to implement
+            /// evaluate_all or evaluate_all_with_rank, if possible
+            fn evaluate(
+                &mut self,
+                values: &[ArrayRef],
+                range: &std::ops::Range<usize>,
+            ) -> Result<ScalarValue> {
+                // Again, the input argument is an array of floating
```
Comment (alamb) [inline datafusion/proto/src/logical_plan/mod.rs:2883]: Perhaps we can just use `todo!()` here and in the other functions? These should not be called during the logical plan testing

**Review by alamb — COMMENTED** [review]
Thanks @parkma99  -- this is looking really close. I have just one more request on the tests 🙏

**Review by alamb — APPROVED** [review]
Thank you @parkma99  -- looks great

**Pull request merged**